### DISCUSSION TITLE
How to create an empty folder in a zip file

### DISCUSSION THREAD

**Discussion by sane-git** [nmoinvaz/minizip#600] [Q&A] [answered]
Hello,
I'm looking for a way to create an empty folder in a zip file.
Is this possible at all?

The code I'm using is as follows:

```
	CT2A filename(lpszPath, CP_UTF8);
	mz_zip_file file_info = { 0, };

	file_info.version_madeby = MZ_VERSION_MADEBY;
	file_info.compression_method = MZ_COMPRESS_METHOD_DEFLATE;
	file_info.filename = filename.m_psz;
	file_info.uncompressed_size = 0;
	file_info.flag = MZ_ZIP_FLAG_UTF8;
	file_info.aes_version = MZ_AES_VERSION;
	file_info.external_fa = FILE_ATTRIBUTE_DIRECTORY; // CREATE A FOLDER

	void* pStream = nullptr;

	mz_stream_os_create(&pStream);
	if (pStream != nullptr)
	{
		mz_zip_writer_add_info(pWriter,** pStream, mz_stream_read, &file_info);

		if (mz_stream_is_open(pStream) == MZ_OK)
			mz_stream_close(pStream);

		mz_stream_delete(&pStream);
		pStream = nullptr;
	}
```

When opened with 7zip the zip file's content is displayed as follows, where **B** is the folder name given by **filename** (see above):
![Folder-Only](https://user-images.githubusercontent.com/94435184/141972689-afdeadc9-50ed-4231-97f3-3d368c858a54.png)

Unzipping the content to disk seems to work as I expect it, i.e. the folder B is created in the filesystem.
However, in the zip file the folder doesn't a appear to be a folder. The "folder" is white instead of yellow.

But the result gets more strange to  me when I conditionally add a file to this folder:

```
	if (bAddFile)
	{
		CString sFileInZip(_T("B\\Arrow.png"));
		CString sFileFromDisk(_T("C:\\Temp\\Images\\Arrow.png"));

		mz_zip_writer_add_file(pWriter, CW2A(sFileFromDisk, CP_UTF8), CT2A(sFileInZip, CP_UTF8));
	}
```

![Folder-WithFiles](https://user-images.githubusercontent.com/94435184/141976560-d9a4261d-4d45-4fab-b0ed-36922ec386be.png)

Now I have the empty folder **B** (which is white) that has been created first, and another folder **B** (real folder look) that contains the file added to it secondly. When trying to unzip 7zip comes with an error message, that duplicates are contained within the zip file.

My assumption now is, that **mz_zip_writer_add_info** is not the way to go when it comes to create empty folders using minizip.
Is that correct?
Is there a way to create an empty folder inside a zip file, so that 7zip can read and display it as a folder?

Many thanks for looking into this issue in advance.

Best regards,
Sascha

**Comment by sane-git** [comment]
Hello,
I'm still looking for a solution to my problem.

Unfortunately, my english is not the best.
So it might be possible that I didn't make my point clear enough.
It this case, please let me know.

Anyway, I still need your help...

Best regards,
Sascha

**Comment by sane-git** [comment, marked as answer]
Hello again,
I found a solution to my problem.

In my project I used windows style backward slashes (\\) to create an empty folder.
After I changed these windows style backward slashes to unix style forward slash everything was working as expected.

Thanks for a great library.

Best regards,
Sascha

**Comment by Simon-zym** [comment]
mz_zip_writer_add_file   this function must use in windows?
